### PR TITLE
pkgsStatic.openssh: fix build

### PR DESCRIPTION
**Pull request by veprbl**
Remove reference to libredirect for cross builds, as for those the tests are disabled:
https://github.com/NixOS/nixpkgs/blob/4818f809a92b2ef8d8c002c10a29f1080106ee81/pkgs/stdenv/generic/make-derivation.nix#L103

###### Motivation for this change

Closes: #154048

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
cc @KAction

**Comment by veprbl** [comment]
@GrahamcOfBorg build pkgsStatic.openssh

File: pkgs/tools/networking/openssh/common.nix [hunk 1]
```diff
@@ -98,6 +82,8 @@ stdenv.mkDerivation rec {
     ++ optional (!linkOpenssl) "--without-openssl"
     ++ extraConfigureFlags;
 
+  ${if stdenv.hostPlatform.isStatic then "NIX_LDFLAGS" else null}="-lkeyutils -laudit";
+
```
Comment (KAction) [inline pkgs/tools/networking/openssh/common.nix:86]: Nice. Didn't knew about this "null =" trick

Comment (KAction) [reply pkgs/tools/networking/openssh/common.nix:86]: You need conditional on withKerberos, I think

Comment (KAction) [reply pkgs/tools/networking/openssh/common.nix:86]: And I thought NIX_LDFLAGS is generally frown upon.

Comment (veprbl) [reply pkgs/tools/networking/openssh/common.nix:86]: > And I thought NIX_LDFLAGS is generally frown upon.

That is correct. For some `LDFLAGS` is not respected here, so I don't see another way.

Comment (KAction) [reply pkgs/tools/networking/openssh/common.nix:86]: Well, you can sed stuff, like I did in my PR.

**Pull request merged**